### PR TITLE
update sbt-plugins

### PR DESCRIPTION
**Pull request by bbstilson**
- drop bintray for codeartifact
- drop sbt 0.13 support
- drop including plugins in plugin

This was successfully published to CodeArtifact.

### CODE REVIEW

File: project/Release.scala [hunk 1]
```diff
@@ -41,11 +27,11 @@ object Release {
     inquireVersions,
     runClean,
     releaseStepCommandAndRemaining("^test"),
-    releaseStepCommandAndRemaining("^ scripted"),
```
[the remark this answers is not in the document]
Comment (bbstilson) [reply project/Release.scala:44]: I don't think it makes a difference. Just tidying up. Once I get CI running we will know for sure though.

File: project/Release.scala [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-import bintray.BintrayKeys._
 import sbtrelease.ReleasePlugin.autoImport._
 import sbtrelease.ReleaseStateTransformations._
 
 object Release {
 
   def releaseSettings = Seq(
-    crossSbtVersions := Vector("1.3.10", "0.13.16"),
-    organization := "org.allenai.plugins",
+    organization := "org.allenai",
```
Comment (ca16) [inline project/Release.scala:9]: why this change?

Comment (bbstilson) [reply project/Release.scala:9]: consistency. It doesn't functionally change anything, but all of our libraries are under the org.allenai organization.

File: src/main/scala/org/allenai/plugins/CoreRepositories.scala [hunk 3]
```diff
@@ -1,53 +0,0 @@
...
-    val ai2PublicReleases = Resolver.bintrayRepo("allenai", "maven")
-
-    // needed for spray-json:
-    val spray = "spray".at("https://repo.spray.io/")
-
-    val typesafeReleases = "Typesafe Releases".at("https://repo.typesafe.com/typesafe/releases/")
-
-    /** Default set of resolvers that will be added via CoreSettings */
-    val defaults = Seq(
-      spray,
```
Comment (ca16) [inline src/main/scala/org/allenai/plugins/CoreRepositories.scala:26]: Do we not need these other resolvers anymore?

File: project/Release.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbstilson) [reply project/Release.scala:44]: If it does make a difference, then it was never doing anything haha

File: src/main/scala/org/allenai/plugins/CoreRepositories.scala (hunk 3 above)
Comment (bbstilson) [reply src/main/scala/org/allenai/plugins/CoreRepositories.scala:26]: Spray and typesafe, no. And I couldn't find anything that used jcenter. It's trivial to add back, and if we need them I will be adding them in scholar most likely.

**Comment by ca16** [comment]
> drop sbt 0.13 support

This means:
- Anything still using sbt 0.13 will be able to keep reading from Bintray until Bintray goes away
- Once Bintray goes away, anything still using sbt 0.13 will have to upgrade so it can use the CodeArtifact version which won't support 0.13 (though hopefully this would happen before Bintray goes away)
- Anything not using sbt 0.13 will be able to keep reading from Bintray until Bintray goes away, or switch to CodeArtifact whenever

File: src/main/scala-sbt-0.13/org/allenai/plugins/Compat.scala [hunk 4]
```diff
@@ -1,22 +0,0 @@
-package org.allenai.plugins
```
Comment (ca16) [inline src/main/scala-sbt-0.13/org/allenai/plugins/Compat.scala:1]: These two Compat files won't be necessary anymore because we will no longer be publishing across sbt versions?

Comment (bbstilson) [reply src/main/scala-sbt-0.13/org/allenai/plugins/Compat.scala:1]: Correct

File: project/Release.scala [hunk 5]
```diff
@@ -41,11 +27,11 @@ object Release {
     inquireVersions,
     runClean,
     releaseStepCommandAndRemaining("^test"),
-    releaseStepCommandAndRemaining("^ scripted"),
+    releaseStepCommandAndRemaining("^scripted"),
     setReleaseVersion,
     commitReleaseVersion,
     tagRelease,
-    releaseStepCommandAndRemaining("^publish"),
+    releaseStepCommandAndRemaining("^codeArtifactPublish"),
```
[the remark this answers is not in the document]
Comment (bbstilson) [reply project/Release.scala:34]: They wouldn't even necessarily need to fully switch, just add the plugin, which adds the resolver.

File: src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala [hunk 6]
```diff
@@ -783,7 +785,7 @@ $DOCKERFILE_SIGIL
 
     // Verify the dockerfile.
     if (verifyDockerfileOnBuild.value) {
-      val isUnchanged = checkDockerfile.value
+      val isUnchanged = checkDockerfile.value: @sbtUnchecked
```
Comment (ca16) [inline src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala:788]: so did this used to not need `@sbtUnchecked` but now needs it?

File: project/Release.scala [hunk 7]
```diff
@@ -18,19 +16,7 @@ object Release {
         "git@github.com:allenai/sbt-plugins.git"
       )
     ),
-    // Publish settings.
-    // http://www.scala-sbt.org/0.12.2/docs/Detailed-Topics/Artifacts.html
-    publishArtifact := true, // Enable publish
     publishArtifact in Test := false,
-    publishMavenStyle := false,
```
Comment (ca16) [inline project/Release.scala:25]: Do you know if the default is false off the top of your head?
Looking at https://www.scala-sbt.org/1.x/docs/Publishing.html#Modifying+the+generated+POM I would _guess_ it is false but not sure at all.

Comment (bbstilson) [reply project/Release.scala:25]: It's set by the codeartifact plugin.

File: src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala (hunk 6 above)
Comment (bbstilson) [reply src/main/scala/org/allenai/plugins/DockerBuildPlugin.scala:788]: it was warning before, but this explicitly silences the warning because the publish step was exiting on warnings. That's also why I moved some of the loggers in here around.

File: project/Release.scala (hunk 7 above)
Comment (ca16) [reply project/Release.scala:25]: So it looks like we're changing the setting right?
https://github.com/bbstilson/sbt-codeartifact/blob/397b52921f6fbfd22e6fc2923c14232dd1878069/sbt-codeartifact/src/main/scala/codeartifact/CodeArtifactPlugin.scala#L49

Is this a CodeArtifact requirement or something else? Not objecting to it, just trying to understand it.

File: src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala [hunk 8]
```diff
@@ -95,7 +95,7 @@ object VersionInjectorPlugin extends AutoPlugin {
   val gitSha1Task = gitSha1 := (gitCommand("rev-parse", "HEAD").!!).trim
 
   val gitRemotesTask = gitRemotes := {
-    val remotes = gitCommand("remote").lines.toList
+    val remotes = gitCommand("remote").lineStream.toList
```
Comment (ca16) [inline src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala:98]: why?

**Comment by ca16** [comment]
>This was successfully published to CodeArtifact.

Is it relatively easy to get a local version of something that depends on sbt-plugins to add that published version and see if anything obvious breaks? Or does CI do something similar?

**Comment by bbstilson** [comment]
>Is it relatively easy to get a local version of something that depends on sbt-plugins to add that published version and see if anything obvious breaks? 

`publishLocal`!

File: src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala (hunk 8 above)
Comment (bbstilson) [reply src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala:98]: deprecation warning haha

Comment (bbstilson) [reply src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala:98]: I just fix things when I see them

**Comment by ca16** [comment]
> Is it relatively easy to get a local version of something that depends on sbt-plugins to add that published version and see if anything obvious breaks?

> publishLocal!

I think I didn't phrase my question correctly... I was thinking more like rn, to see if this new version of sbt-plugins breaks anything obvious, have you tried/is it easy to try running something in scholar (or some other project that uses sbt-plugins) using this version of sbt-plugins? (whether its the one published to CodeArtifacts or one published by `publishLocal`)

File: src/sbt-test/sbt-plugins/simple/project/project/plugins.sbt [hunk 9]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("io.github.bbstilson" % "sbt-codeartifact" % "0.1.2")
```
Comment (ca16) [inline src/sbt-test/sbt-plugins/simple/project/project/plugins.sbt:1]: project/project like you were saying

**Comment by bbstilson** [comment]
@ca16 here's a PR using the changes in this PR: https://github.com/allenai/scholar/pull/27109

**Review by ca16 — APPROVED** [review]
Assuming `publishMavenStyle` works out (whether you keep it as true or not), LGTM.
